internal/zinc-core/src/main/scala/sbt/internal/inc/IncrementalCommon.scala

```diff
@@ -443,17 +443,21 @@ private[inc] abstract class IncrementalCommon(
       isScalaClass: String => Boolean
   ): Set[String] = {
     val initial = changes.allModified.toSet
-    val dependsOnClass = findClassDependencies(_, relations)
+
     val firstClassInvalidation: Set[String] = {
       val invalidated = if (invalidateTransitively) {
         // Invalidate by brute force (normally happens when we've done more than 3 incremental runs)
-        IncrementalCommon.transitiveDeps(initial, log)(dependsOnClass)
+        val dependsOnClass = relations.memberRef.internal.reverse _
```

Comment:
> In #898, transitive invalidation was refactored incorrectly to have the same `dependsOnClass` function value. This restores that.

The definition of `findClassDependencies` is:

```scala
  override protected def findClassDependencies(
      className: String,
      relations: Relations
  ): Set[String] = relations.memberRef.internal.reverse(className)
```

So what is restored?

I don't see how this can change anything and the "Reproduce #918" commit doesn't seem to reproduce a failure.

Reply:
You might be right here though. I can no longer reproduce the fixed effect. It's possible I was testing a wrong Zinc 1.4.0-SNAPSHOT since this was my second attempt.

Reply:
I pushed my first attempt, which still works but reverts the entire #898.